### ISSUE TITLE
Rendering of templated files is different when test=true in 2016.3.3

### ISSUE THREAD

**Issue by luitzifa** [saltstack/salt#37462]
### Description of Issue
**Note: This Bug does not affect 2016.11.0rc1**
When i use file.managed the rendered file differ from expected result and result shown in dry run:
```
root@test:~# salt test state.highstate test=true
test:
----------
          ID: /tmp/testfile2
    Function: file.managed
      Result: True
     Comment: The file /tmp/testfile2 is in the correct state
     Started: 11:15:07.787620
    Duration: 44.257 ms
     Changes:   
----------
          ID: /tmp/testfile1
    Function: file.managed
      Result: None
     Comment: The file /tmp/testfile1 is set to be changed
     Started: 11:15:07.831991
    Duration: 15.093 ms
     Changes:   
              ----------
              diff:
                  --- 
                  +++ 
                  @@ -13,3 +13,5 @@
                   foo3
                   foo4 = 124
                   
                  +foobar1
                  +foobar2

Summary for test
------------
Succeeded: 2 (unchanged=1, changed=1)
Failed:    0
------------
Total states run:     2
Total run time:  59.350 ms



root@test:~# salt test state.highstate
test:
----------
          ID: /tmp/testfile2
    Function: file.managed
      Result: True
     Comment: File /tmp/testfile2 is in the correct state
     Started: 11:15:16.149654
    Duration: 118.947 ms
     Changes:   
----------
          ID: /tmp/testfile1
    Function: file.managed
      Result: True
     Comment: File /tmp/testfile1 is in the correct state
     Started: 11:15:16.268718
    Duration: 28.557 ms
     Changes:   

Summary for test
------------
Succeeded: 2
Failed:    0
------------
Total states run:     2
Total run time: 147.504 ms

```

### Setup
You can use this repository to reproduce:
https://github.com/luitzifa/salt-testrun-templating-bug

### Versions Report
```
root@test:~# salt --versions-report
Salt Version:
           Salt: 2016.3.3
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.3.7
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Jul  1 2016, 15:12:24)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.4.0-45-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by Ch3LL** [comment]
@luitzifa i just tested this and it seems to be fixed in 2016.3.4. Can you confirm that is also true for you?

**Comment by luitzifa** [comment]
@Ch3LL Yes, it's fixed in 2016.3.4   Thank you
